src/Source.jl

```diff
@@ -235,7 +239,7 @@ Keyword Arguments:
 * `types`: column types can be provided manually as a complete Vector{Type}, or in a Dict to reference individual columns by name or number
 * `nullable::Bool`: indicates whether values can be nullable or not; `true` by default. If set to `false` and missing values are encountered, a `Data.NullException` will be thrown
 * `footerskip::Int`: indicates the number of rows to skip at the end of the file
-* `rows_for_type_detect::Int=100`: indicates how many rows should be read to infer the types of columns
+* `rows_for_type_detect::Int=20`: indicates how many rows should be read to infer the types of columns
```

Comment:
I hadn't noticed this. I realize this is just a fix to match the actual behavior, but I think we should raise this value. Something like 100 or 200 sounds safer, and shouldn't impact performance a lot, should it @quinnj?

Reply:
IIUC most of the time is spent compiling methods for the new scheme.
I've tried optimizing the `rows_for_type_detect` for johnmyleswhite/RDatasets.jl#48, but that didn't really improve the test time (~5 min for 733 datasets).